src/test/java/io/lenses/connect/smt/header/TimestampConverterTest.java

```diff
@@ -883,6 +886,66 @@ void testWithSchemaValuePrefixedFieldConversion_Seconds() {
     assertEquals(expectedDate.getTime(), header.value());
   }
 
+  @Test
+  void testWithRecordMetadataPrefixedFieldConversion_Seconds() {
+    Map<String, String> config = new HashMap<>();
+    config.put(TimestampConverter.TARGET_TYPE_CONFIG, "Timestamp");
+    config.put(TimestampConverter.FIELD_CONFIG, "_timestamp");
+    config.put(TimestampConverter.UNIX_PRECISION_CONFIG, "seconds");
+    config.put(TimestampConverter.HEADER_NAME_CONFIG, "ts_header");
+    // ts field is a unix timestamp with seconds precision
+    Schema structWithTimestampFieldSchema =
+        SchemaBuilder.struct().field("ts", Schema.INT64_SCHEMA).build();
+    Struct original = new Struct(structWithTimestampFieldSchema);
+    original.put("ts", DATE_PLUS_TIME_UNIX_SECONDS);
+
+    final TimestampConverter<SourceRecord> transformer = new TimestampConverter<>();
+    transformer.configure(config);
+    final SourceRecord initial =
+        new SourceRecord(
+            null,
+            null,
+            "topic",
+            0,
+            null,
+            null,
+            structWithTimestampFieldSchema,
+            original,
+            DATE_PLUS_TIME_UNIX_SECONDS);
+    final SourceRecord transformed = transformer.apply(initial);
+
+    Calendar expectedDate = GregorianCalendar.getInstance(UTC);
+    expectedDate.setTimeInMillis(0L);
+    expectedDate.add(Calendar.DATE, 1);
+    expectedDate.add(Calendar.SECOND, 1);
```

Comment:
it's generally recommended to use the new Date/Time API (java.time) introduced in Java 8 instead of GregorianCalendar whenever possible.